vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1057,32 +1062,54 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Invalid disks to set: %#v", disks)
 	}
 
+	// start
 	networkInterfaces := make([]map[string]interface{}, 0)
+
+	deviceList, err := vm.Device(context.TODO())
```

Comment:
@mjrider just a note here - am I right in assuming that `err` is not being caught here?

Also = I notice this device list is being sourced off of the `vm` MO instead of the `mvm` MO, although the device being logged in debug is being sourced off of `mvm`. Is this an error or by design?